File: app/src/main/java/com/lambdaschool/congressdata/CongresspersonListViewModel.java
package com.lambdaschool.congressdata;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

import java.util.ArrayList;

public class CongresspersonListViewModel extends ViewModel {
    private LiveData<ArrayList<OfficialOverview>> overviewList;

    public LiveData<ArrayList<OfficialOverview>> getOverviewList() {
        if (overviewList == null) {
            loadList();
        }
        return overviewList;
    }

    private void loadList() {
        overviewList = OverviewListRepository.getOverviewList();
    }
}

File: app/src/main/java/com/lambdaschool/congressdata/CongresspersonListViewModel.kt
package com.lambdaschool.congressdata

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.ViewModel

import java.util.ArrayList

class CongresspersonListViewModel : ViewModel() {
    private var overviewList: LiveData<ArrayList<OfficialOverview>>? = null

    fun getOverviewList(): LiveData<ArrayList<OfficialOverview>>? {
        if (overviewList == null) {
            loadList()
        }
        return overviewList
    }

    private fun loadList() {
        overviewList = OverviewListRepository.overviewList
    }
}
